src/index.ts

```diff
@@ -452,21 +467,31 @@ export default class Wallet {
    * @param input A JSON serialized string, or an object representing EthSale Keystore.
    * @param password The keystore password.
    */
-  public static fromEthSale(input: string | EthSaleKeystore, password: string): Wallet {
+  public static async fromEthSale(
+    input: string | EthSaleKeystore,
+    password: string
+  ): Promise<Wallet> {
     const json: EthSaleKeystore = typeof input === 'object' ? input : JSON.parse(input)
 
     const encseed = Buffer.from(json.encseed, 'hex')
 
     // key derivation
-    const derivedKey = crypto.pbkdf2Sync(password, password, 2000, 32, 'sha256').slice(0, 16)
+    const pass = Buffer.from(password, 'utf8')
+    const derivedKey = pbkdf2Sync(pass, pass, 2000, 32, 'sha256').slice(0, 16)
 
     // seed decoding (IV is first 16 bytes)
     // NOTE: crypto (derived from openssl) when used with aes-*-cbc will handle PKCS#7 padding internally
     //       see also http://stackoverflow.com/a/31614770/4964819
-    const decipher = crypto.createDecipheriv('aes-128-cbc', derivedKey, encseed.slice(0, 16))
-    const seed = runCipherBuffer(decipher, encseed.slice(16))
+    const seed = await aes.decrypt(
+      encseed.slice(16),
```

Comment:
VSCode is complaining about this since this is a `Buffer.prototype.slice` vs `Uint8Array.prototype.slice` which is what `aes.decrypt` calls for.  It's a minor point but should we use the correct `slice` method here (and potentially convert?